cms/tests/test_toolbar.py

```diff
@@ -1865,6 +1874,119 @@ def test_edit_page(self):
+
+    @override_settings(CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM_ENABLED=True)
+    @override_settings(CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM="test-live-link")
+    @patch("cms.utils.urlutils.admin_reverse")
+    def test_add_live_url_querystring_param_with_querystring(self, patched_admin_reverse):
+        """
+        With the endpoint returning an existing querystring param, the additional param should be appended
+        to the existing with &.
+        """
+        page = create_page("home", 'nav_playground.html', "en")
+        page_content = page.get_title_obj()
+        app_label = page_content._meta.app_label
+        model_name = page_content._meta.model_name
+        live_url = page.get_absolute_url()
+        content_type = ContentType.objects.get(app_label=app_label, model=model_name)
+        # Get the original endpoint url, and patch it with additional querystring parameter
+        base_url = admin_reverse('cms_placeholder_render_object_edit', args=[content_type.pk, page_content.pk])
+        patched_admin_reverse.return_value = f"{base_url}?base_qsp=base_value"
+
+        with patch.object(utils, "admin_reverse", return_value=f"{base_url}?base_qsp=base_value"):
+            edit_url = get_object_edit_url(page_content)
+
+        self.assertIn(f"?base_qsp=base_value&test-live-link={live_url}", edit_url)
+        self.assertEqual(edit_url.count("?"), 1)
+        self.assertEqual(edit_url.count("&"), 1)
+
+    @override_settings(CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM="test-live-link")
+    def test_add_live_url_querystring_param_no_querystring_setting_disabled(self):
+        """
+        With the querystring param configured, but CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM_ENABLED not set True,
+        don't add the querystring params
+        """
+        page = create_page("home", 'nav_playground.html', "en")
+        page_content = page.get_title_obj()
+        content_type = ContentType.objects.get_for_model(page_content)
+        language = get_language()
+        with override(language):
+            expected_edit_url = admin_reverse(
+                'cms_placeholder_render_object_edit', args=[content_type.pk, page_content.pk]
+            )
+            expected_preview_url = admin_reverse(
+                'cms_placeholder_render_object_preview', args=[content_type.pk, page_content.pk]
+            )
+
+        edit_url = get_object_edit_url(page_content)
+        preview_url = get_object_preview_url(page_content)
+
+        self.assertEqual(edit_url, expected_edit_url)
+        self.assertEqual(preview_url, expected_preview_url)
+        self.assertEqual(edit_url.count("?"), 0)
+        self.assertEqual(preview_url.count("?"), 0)
+        self.assertEqual(edit_url.count("&"), 0)
+        self.assertEqual(preview_url.count("&"), 0)
+
+    @override_settings(CMS_ENDPOINT_LIVE_URL_QUERYSTRING_PARAM_ENABLED=True)
+    def test_add_live_url_querystring_param_handles_missing_enabled_setting(self):
```

Comment:
Not required, this was part of the older logic I think where the configuration was required. The param has a default, currently you can't use that default because it's enforced that it's set.